osu.Game/Database/EmptyRealmSet.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using Realms;
+using Realms.Schema;
+
+#nullable enable
+
+namespace osu.Game.Database
+{
+    public class EmptyRealmSet<T> : IRealmCollection<T>
```

Comment:
Slightly unfortunate, but can't find a better way (realm collection classes have `internal` constructors).

Reply:
On a second look the locking indeed shouldn't be required. Just a bit weird 1/3 acceses are under the lock, looks like a bug in most circumstances.